### PR TITLE
Authorization Code Flow for Single-Page Applications: Samples and Docs Updates Round 1

### PR DESCRIPTION
**Pull request by pkanher617**
Please review #1320 before reviewing this PR.

This PR introduces new and improved samples (model was taken from [sample repo](https://github.com/Azure-Samples/active-directory-javascript-graphapi-v2)) for 1.0 and 2.0 and updated READMEs for the msal-common and msal-browser packages.

It also includes placeholders for additional documentation in the 2.0 version of the library.

### CODE REVIEW

File: lib/msal-browser/README.md [hunk 1]
```diff
@@ -1,13 +1,60 @@
...
+
+*Expect us to detail our major and minor releases moving forward, while leaving out our patch releases.  Patch release notes can be found in our change log.*
+
+Date | Release | Announcement | Main features
+------| ------- | ---------| ---------
+January 17, 2020 | msal-browser v2.0.0-alpha | No release notes yet | Alpha version of the msal-browser package with authorization code flow for SPAs working in dev; relies on msal-common v1.0.0-alpha
+
+## Pre-requisites
+
+- MSAL.js is meant to be used in [Single-Page Application scenarios](https://docs.microsoft.com/en-us/azure/active-directory/develop/scenario-spa-overview).
```
Comment (hamiltonha) [inline lib/msal-browser/README.md]: you need to remove the "en-us" in links so they localize

File: lib/msal-browser/README.md [hunk 2]
```diff
@@ -1,13 +1,60 @@
...
+
+- Before using MSAL.js you will need to [register an application in Azure AD](https://docs.microsoft.com/en-us/azure/active-directory/develop/scenario-spa-app-registration) to get a valid `clientId` for configuration, and to register the routes that your app will accept redirect traffic on.
+
+1. Create a new application registration in the portal. Use whatever audience you wish, as long as it is testable on your machine. It is recommended to use the common audience (a.k.a. accounts in any Azure tenant) for simplest use.
```
Comment (hamiltonha) [inline lib/msal-browser/README.md]: updated registration steps: 
- go to the authentication blade
-click "add a platform"
-click "single-page app" tile 
-enter a redirect URI, do not check any implicit settings 
-in the authentication blade, under advanced settings, set "yes" for "treat application as public client"

File: lib/msal-browser/README.md (hunk 1 above)
Comment (pkanher617) [reply lib/msal-browser/README.md]: done.

File: lib/msal-browser/README.md (hunk 2 above)
Comment (pkanher617) [reply lib/msal-browser/README.md]: done.

**Pull request merged**